### PR TITLE
Feature/coverage

### LINKED ISSUE #148: Visualizer - Add Support For Code Coverage Checking
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
We should add support for code coverage checking to the typescript code that lives in the visualizer. 

**Describe the solution you'd like**
One each build, Travis CI should automatically upload code coverage information to codecov. Here is an example of how to do this for typescript: https://github.com/codecov/example-typescript

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description
Travis CI is now added for testing coverage. 

### Testing Done
Running `yarn test:coverage` shows that test coverage is integrated.

### Resolved Issues
Resolves #148 

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

***Feel free to make additions of things that we should be checking to this file if you think there's something missing!!!!***

### CODE REVIEW

**Comment by qgolsteyn** [comment]
@kelvinkoon Could you run `yarn` in this branch and push the resulting changes. This should fix the problem.

**Comment by kelvinkoon** [comment]
> @kelvinkoon Could you run `yarn` in this branch and push the resulting changes. This should fix the problem.

Just finished running `yarn` to fix the lockfile. Changes have been pushed.

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,13 @@ sudo: false
 dist: trusty
 services:
   - docker
-language: generic
+language:
```
Comment (qgolsteyn) [inline .travis.yml]: This whole set of changes is not needed. Gareth made a script that does this for us

File: src/corner_kick/package.json [hunk 2]
```diff
@@ -22,6 +23,7 @@
         "@types/roslib": "0.18.3",
         "@types/styled-components": "4.1.4",
         "clean-webpack-plugin": "1.0.0",
+        "codecov": "3.1.0",
```
Comment (qgolsteyn) [inline src/corner_kick/package.json]: This isn't needed either thanks to Gareth's script

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
There is already a script to capture coverage once it is generated, so I added some comments to highlight areas of the code that are no longer necessary.

**Comment by qgolsteyn** [comment]
Will need to fix permissions on test scripts

File: .travis.yml (hunk 1 above)
Comment (kelvinkoon) [reply .travis.yml]: Removed the set of changes.

File: src/corner_kick/scripts/start_test.sh [hunk 3]
```diff
@@ -7,4 +7,5 @@ DIRECTORY=`dirname $0`
 # Move to this directory and start the test suite of
 # Corner Kick.
 cd $DIRECTORY
-yarn test
+chmod+x
```
Comment (qgolsteyn) [inline src/corner_kick/scripts/start_test.sh]: No. That's not how you fix permissions. You need to run `chmod +x start_test.sh` ON the file, not write it IN the file. From the scripts directory, open a console and run `chmod +x start_test.sh` then commit.

**Review by qgolsteyn — CHANGES_REQUESTED** [review]

File: src/corner_kick/scripts/start_test.sh (hunk 3 above)
Comment (kelvinkoon) [reply src/corner_kick/scripts/start_test.sh]: Fixed this one. 
Is the command for pulling from upstream "git fetch upstream"?

**Comment by MathewMacDougall** [comment]
CI is still complaining `error Command "test:coverage" not found.`

File: .travis.yml [hunk 4]
```diff
@@ -17,35 +17,35 @@ notifications:
...
-      --env RUN_COVERAGE
-      --env RUN_FORMATTING_CHECKS
-      --env ROS_DISTRO 
-      --env REPOSITORY_NAME
-      --env TRAVIS_PULL_REQUEST
-      --env TRAVIS_BRANCH
-      --volume $(pwd):/root/$REPOSITORY_NAME
-      $DOCKER_IMAGE
-      /bin/bash -c "/root/$REPOSITORY_NAME/travis_ci.sh"
+  # Pull in MoveIt's ROS Docker CI image (piping to `/dev/null` to hide output)
```
Comment (qgolsteyn) [inline .travis.yml]: This file should not be modified.

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
It appear there are no modifications to `package.json` in this PR, which would explain the CI error. We should add a new script in `package.json` for `test:coverage`.

**Review by qgolsteyn — APPROVED** [review]
LGTM :+1:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**